apps/freeablo/faworld/actor/attackstate.cpp

```diff
@@ -1,47 +1,107 @@
 #include "attackstate.h"
 #include "../../fasavegame/gameloader.h"
 #include "../actor.h"
+#include "../spells.h"
 
 namespace FAWorld
 {
-
     namespace ActorState
     {
-        const std::string MeleeAttackState::typeId = "actorstate-attack-state";
+        const std::string BaseAttackState::typeId = "actorstate-base-attack-state";
```

Comment:
BaseAttackState is already an abstract class, so it doesn't need to have an id. It's getTypeId() method can be left unimplemented.
This is only necessary so we can tell what class to use for deserialisation, and a BaseAttackState can never be saved anyway since it can't be instantiated directly (without using a subclass)